### ISSUE TITLE
Unexpected randomness in graphconv prediction

### ISSUE THREAD

**Issue by mc-robinson** [deepchem/deepchem#1506]
VERSION: 2.1.0

After training a GraphConvModel for a fair number of hours, I am getting some unexpected results when testing the performance of the model. In particular, the training, valid, and test errors are all different when I run the code below. This is surprising since I was careful to set the same seed during the train test split and have not changed the csv file containing the data between training and testing. 

If it helps, my original model was constructed using the following code:

TRAINING CODE:
```
model = GraphConvModel(n_tasks=1, mode='regression',
                       tensorboard=True, model_dir='models/',
                       dropout=0.5, graph_conv_layers=[64,64,64])
```
Is it possible that dropout is being used in the prediction, leading to the stochastic behaviour? Otherwise, I cannot find anywhere else that randomness has been introduced, and this is quite frustrating since I did not save my datasets to disk, thinking that setting the seed would be sufficient. I have all of the model data, but I doubt that helps without knowing if I am snooping on my test data.

TESTING CODE:
```
import deepchem as dc
from deepchem.models.tensorgraph.models.graph_models import GraphConvModel

graph_featurizer = dc.feat.graph_features.ConvMolFeaturizer()
loader = dc.data.data_loader.CSVLoader( tasks=['unique_pka'],
                                         smiles_field="CANONICAL_SMILES",
                                        id_field="CMPD_CHEMBLID",
                                        featurizer=graph_featurizer )

dataset = loader.featurize( './unique_pkas.csv' )

splitter = dc.splits.RandomSplitter()
train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(dataset, seed=42)

transformers = [
        dc.trans.NormalizationTransformer(transform_y=True, dataset=train_dataset)]

for dataset in [train_dataset, valid_dataset, test_dataset]:
        for transformer in transformers:
                    dataset = transformer.transform(dataset)

model = GraphConvModel.load_from_dir('models')
# model.restore()

train_scores = model.evaluate(
                train_dataset,
                [dc.metrics.Metric(dc.metrics.rms_score),
                 dc.metrics.Metric(dc.metrics.r2_score)]
                )
print('train scores')
print(train_scores)

valid_scores = model.evaluate(
                valid_dataset,
                [dc.metrics.Metric(dc.metrics.rms_score),
                 dc.metrics.Metric(dc.metrics.r2_score)]
                )
print('valid scores')
print(valid_scores)

test_scores = model.evaluate(
                test_dataset,
                [dc.metrics.Metric(dc.metrics.rms_score),
                 dc.metrics.Metric(dc.metrics.r2_score)]
                )
print('test_scores')
print(test_scores)

```
Thanks for your help!

**Comment by peastman** [comment]
> In particular, the training, valid, and test errors are all different when I run the code below.

Different from what?  From each other?  Or different between multiple runs of the script?  Or something else?

**Comment by mc-robinson** [comment]
Sorry. Different as in the test error changes between multiple runs of the script, despite the seed being set each time. This is also true of the training and validation set errors. This confuses me because I set the seed in order to try to maintain reproducibility. 

They are also different from each other, as is expected.

**Comment by peastman** [comment]
Ok, let me look into it and see if I can tell what's going on.

**Comment by mc-robinson** [comment]
As an update, I tried splitting the train, validation, and test sets in a separate pandas script. I then loaded each dataset individually into deepchem in order to ensure the train/valid/test split was the same each time. Now the code outputs the same results each time and the results are much more what one would expect, indicating that I am not snooping on the test set anymore. Therefore, my instinct is that the aforementioned behavior may be due to the splitter function in deepchem?

For reference the new script is posted below:

```
import deepchem as dc
from deepchem.models.tensorgraph.models.graph_models import GraphConvModel

graph_featurizer = dc.feat.graph_features.ConvMolFeaturizer()
loader = dc.data.data_loader.CSVLoader( tasks=['unique_pka'],
                                         smiles_field="CANONICAL_SMILES",
                                        id_field="CMPD_CHEMBLID",
                                        featurizer=graph_featurizer )

# dataset = loader.featurize( './unique_pkas.csv' )

# splitter = dc.splits.RandomSplitter()
# train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(dataset, seed=42)

train_dataset = loader.featurize( './unique_pkas_train.csv' )
valid_dataset = loader.featurize( './unique_pkas_valid.csv' )
test_dataset = loader.featurize( './unique_pkas_test.csv' )

transformers = [
        dc.trans.NormalizationTransformer(transform_y=True, dataset=train_dataset)]

for dataset in [train_dataset, valid_dataset, test_dataset]:
        for transformer in transformers:
                    dataset = transformer.transform(dataset)

model = GraphConvModel.load_from_dir('models')
# model.restore()

train_scores = model.evaluate(
                train_dataset,
                [dc.metrics.Metric(dc.metrics.rms_score),
                 dc.metrics.Metric(dc.metrics.r2_score),
                 dc.metrics.Metric(dc.metrics.mae_score)]
                )
print('train scores')
print(train_scores)

valid_scores = model.evaluate(
                valid_dataset,
                [dc.metrics.Metric(dc.metrics.rms_score),
                 dc.metrics.Metric(dc.metrics.r2_score),
                 dc.metrics.Metric(dc.metrics.mae_score)]
                )
print('valid scores')
print(valid_scores)

test_scores = model.evaluate(
                test_dataset,
                [dc.metrics.Metric(dc.metrics.rms_score),
                 dc.metrics.Metric(dc.metrics.r2_score),
                 dc.metrics.Metric(dc.metrics.mae_score)]
                )
print('test_scores')
print(test_scores)
```

**Comment by peastman** [comment]
You're completely right.  `train_test_valid_split()` was ignoring the seed instead of passing it on to the splitter.  I just posted a pull request with the fix.  Thanks for spotting this!

**Comment by mc-robinson** [comment]
Glad you found the error! I really hope others had noticed this or were not using the seed for sake of reproducibility, because it would very easily lead one to have an overly optimistic opinion of their generalization performance...
